massa-execution/src/vm.rs

```diff
@@ -84,9 +85,12 @@ impl VM {
         if let Some(cached) = self.is_already_done(step) {
             // execution was already done, apply cached ledger changes to final ledger
             // It should always happen
-            let context = CONTEXT.lock().unwrap();
-            let mut final_ledger_guard = context.ledger_step.final_ledger_slot.lock().unwrap();
-            final_ledger_guard.0.apply_changes(&cached);
+            let mut context = self.execution_context.lock().unwrap();
+            (*context)
+                .ledger_step
+                .final_ledger_slot
+                .ledger
+                .apply_changes(&cached);
         }
     }
```

Comment:
looks like the recursive call that we used to have here and that was removed because of some temporary async attempt was never put back after the revert !

Reply:
Ok, can you open an issue to fix this please?

Reply:
looks like it's been fixed elsewhere: https://github.com/massalabs/massa/blob/feature/execution/add_cautious_bassoon/massa-execution/src/vm.rs#L85-L91

... and in a smarter way than the one I was thinking about :)